spec/serialization_spec.rb

```diff
@@ -43,6 +45,505 @@ def self.name
+        end
+
+        it 'coerces Time to correct date string format' do
+          time = Time.parse('2018/01/06 4:13pm')
+          PORO::Employee.create(age: time)
+          render
+          expect(attributes['age']).to eq('2018-01-06')
+        end
+
+        it 'coerces strings to date format' do
+          PORO::Employee.create(age: '2018/01/06')
+          render
+          expect(attributes['age']).to eq('2018-01-06')
+        end
+
+        it 'allows nils' do
+          PORO::Employee.create(age: nil)
+          render
+          expect(attributes['age']).to eq(nil)
+        end
+
+        context 'when only month' do
+          before do
+            PORO::Employee.create(age: '2018/01')
+          end
+
+          # You don't want this type if you don't want this conversion
+          # Use a different type, or register a new custom type
+          it 'defaults to first of the month' do
+            render
+            expect(attributes['age']).to eq('2018-01-01')
+          end
+        end
+
+        context 'when cannot coerce' do
+          before do
+            PORO::Employee.create(age: '1')
+          end
+
+          it 'raises error' do
+            expect {
+              render
+            }.to raise_error(Dry::Types::ConstraintError)
+          end
+        end
+      end
+
+      # iso8601
+      # If zone is present, it is preserved
+      # If not present, falls back to UTC
+      context 'when datetime' do
+        before do
+          resource.attribute :age, :datetime
+        end
+
+        # NB Time has an implicit zone based on system time
+        # Here we are setting the zone explicitly
+        it 'coerces Time correctly' do
+          time = Time.parse('2018-01-06 4:36pm PST')
+          PORO::Employee.create(age: time)
+          render
+          expect(attributes['age']).to eq('2018-01-06T16:36:00-08:00')
+        end
+
+        # Default zone UTC
+        # NB: Internal custom type required for this
+        it 'coerces Date correctly' do
+          date = Date.parse('2018-01-06')
+          PORO::Employee.create(age: date)
+          render
+          expect(attributes['age']).to eq('2018-01-06T00:00:00+00:00')
+        end
+
+        it 'preserves time zones on Date' do
+          date_time = DateTime.parse('2018-01-06 4:36pm PST')
+          PORO::Employee.create(age: date_time)
+          render
+          expect(attributes['age']).to eq('2018-01-06T16:36:00-08:00')
+        end
+
+        # No zone, defaults to UTC
+        # NB: Internal custom type required for this
+        it 'coerces DateTime correctly' do
+          date_time = DateTime.parse('2018-01-06 4:36pm')
+          PORO::Employee.create(age: date_time)
+          render
+          expect(attributes['age']).to eq('2018-01-06T16:36:00+00:00')
+        end
+
+        # NB: Internal custom type required for this
+        it 'preserves DateTime zones' do
+          date_time = DateTime.parse('2018-01-06 4:36pm PST')
```

Comment:
How is this test different from line 287?

Reply:
Yeah looks like the same good catch